### PR TITLE
feat: change to manage versions by JSON file in installation directory

### LINKED ISSUE #45: Manage versions by JSON file in installation directory
## Is your feature request related to a problem? Please describe.

If I change the installation path, the version data will be reset even though I have installed the package.

## Describe the solution you'd like

Make `apm.json` in the installation path and store version data in the file.

### PR DESCRIPTION
**Pull request by hal-shu-sato**
## Description

Change to manage versions by JSON file in installation directory
Now you can keep your data even if you change the installation path.

## Related Issue

Close #45 

## Motivation and Context

It is difficult to use because the version information is reset when the installation path is changed.

## How Has This Been Tested?

OS: Windows 10 Home
Version: 21H1

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by hal-shu-sato** [comment]
I will do the rest of the work after #57 is closed.

File: src/lib/apmJson.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+   *
+   * @param {string} instPath - An installation path
+   * @param {object} plugin - An information of a plugin
+   */
+  addPlugin: function (instPath, plugin) {
+    const apmJson = getApmJson(instPath);
+    apmJson.plugins[plugin.id] = {
+      repository: plugin.repo,
+      version: plugin.latestVersion,
+    };
```
Comment (mitosagi) [inline src/lib/apmJson.js:144]:
```suggestion
    apmJson.plugins[plugin.repo + plugin.id] = {
      id: plugin.id,
      repository: plugin.repo,
      version: plugin.latestVersion,
    };
```

Comment (hal-shu-sato) [reply src/lib/apmJson.js:144]: Do you think this kind of change is acceptable?

```suggestion
    apmJson.plugins[plugin.repo][plugin.id] = {
      id: plugin.id,
      repository: plugin.repo,
      version: plugin.latestVersion,
    };
```

Comment (mitosagi) [reply src/lib/apmJson.js:144]: Your way is better.

Comment (hal-shu-sato) [reply src/lib/apmJson.js:144]: Please wait a moment.
I think there is only one plugin/script that can be installed in the same directory but can duplicate IDs happen?

Comment (mitosagi) [reply src/lib/apmJson.js:144]: Sorry, the duplication only occurs when reverse lookup is done from the installation information, so as you pointed out, there is no need to take action here.

Comment (hal-shu-sato) [reply src/lib/apmJson.js:144]: It's okay.
So, for the convenience, I will just add the id property.

```suggestion
    apmJson.plugins[plugin.id] = {
      id: plugin.id,
      repository: plugin.repo,
      version: plugin.latestVersion,
    };
```

**Comment by hal-shu-sato** [comment]
We've made a lot of progress on the commit, so I'll work on the rest after the merge.

**Pull request merged**